src/main/java/com/google/firebase/database/connection/NettyWebSocketClient.java

```diff
@@ -75,20 +80,27 @@
   public void connect() {
     checkState(channel == null, "channel already initialized");
     try {
-      TrustManagerFactory trustFactory = TrustManagerFactory.getInstance(
-          TrustManagerFactory.getDefaultAlgorithm());
-      trustFactory.init((KeyStore) null);
-      final SslContext sslContext = SslContextBuilder.forClient()
-          .trustManager(trustFactory).build();
       Bootstrap bootstrap = new Bootstrap();
+      SslContext sslContext = null;
+      if (this.isSecure) {
+        TrustManagerFactory trustFactory = null;
+        trustFactory = TrustManagerFactory.getInstance(
+            TrustManagerFactory.getDefaultAlgorithm());
+        trustFactory.init((KeyStore) null);
+        sslContext = SslContextBuilder.forClient()
+            .trustManager(trustFactory).build();
+      }
       final int port = uri.getPort() != -1 ? uri.getPort() : DEFAULT_WSS_PORT;
+      final SslContext[] sslContexts = new SslContext[]{sslContext};
```

Comment:
Why are we doing this? Isn't a null-able variable sufficient?

Reply:
I agree. Changing this back to `final SslContext finalSslContext = sslContext` (final-copy) pattern instead of the array. I think this way is cleaner.